rpc/frontend/src/main/scala/io/udash/rpc/internals/ServerConnector.scala

```diff
@@ -27,38 +27,55 @@ abstract class AtmosphereServerConnector[RPCRequest](private val serverUrl: Stri
 
   private val waitingRequests = new mutable.ArrayBuffer[RPCRequest]()
   private var isReady = false
-
-  private val atmRequest = AtmosphereRequest(
-    url = serverUrl,
-    contentType = "application/json",
-    logLevel = "info",
-    transport = Transport.WEBSOCKET,
-    fallbackTransport = Transport.SSE,
-    maxReconnectOnClose = 36000,
-    onOpen = (res: AtmosphereResponse) => ready(true),
-    onReopen = (res: AtmosphereResponse) => ready(true),
-    onError = (res: AtmosphereResponse) => ready(false),
-    onReconnect = (req: AtmosphereRequest, res: AtmosphereResponse) => ready(false),
-    onClose = (res: AtmosphereResponse) => {
-      if (res.request != null) {
-        if (res.request.method == Transport.WEBSOCKET || (res.request.method == Transport.SSE && res.request.method == Method.GET)) {
+  private val websocketSupport: Boolean = scala.scalajs.js.Dynamic.global.WebSocket != null
+
+  private var onReconnectTimeoutHandler = 0
+
+  private val socket: AtmosphereRequest = {
+    if (websocketSupport) {
+      val reconnectInterval = 1000
+      val atmRequest = AtmosphereRequest(
+        url = serverUrl,
+        contentType = "application/json",
+        logLevel = "info",
+        transport = Transport.WEBSOCKET,
+        fallbackTransport = Transport.WEBSOCKET,
+        enableProtocol = true,
+        reconnectInterval = reconnectInterval,
+        maxReconnectOnClose = 36000,
+        onOpen = (res: AtmosphereResponse) => ready(true),
+        onReopen = (res: AtmosphereResponse) => ready(true),
+        onReconnect = (req: AtmosphereRequest, res: AtmosphereResponse) => {
+          if (onReconnectTimeoutHandler != 0) dom.window.clearTimeout(onReconnectTimeoutHandler)
           ready(false)
-        }
-      }
-    },
-    onClientTimeout = (res: AtmosphereResponse) => ready(false),
-    onMessage = (res: AtmosphereResponse) => handleMessage(res.responseBody),
-    onMessagePublished = (req: AtmosphereRequest, res: AtmosphereResponse) => handleMessage(req.responseBody)
-  )
-
-  private var socket = Atmosphere.subscribe(atmRequest)
-
-  timers.setInterval(5000)({
-    if (!isReady) {
-      Atmosphere.unsubscribe()
-      socket = Atmosphere.subscribe(atmRequest)
+          onReconnectTimeoutHandler = dom.window.setTimeout(() => {
+            ready(true)
+            onReconnectTimeoutHandler = 0
+          }, reconnectInterval * 2)
+        },
+        onError = (res: AtmosphereResponse) => ready(false),
+        onClose = (res: AtmosphereResponse) => ready(false),
+        onClientTimeout = (res: AtmosphereResponse) => ready(false),
+        onMessage = (res: AtmosphereResponse) => handleMessage(res.responseBody),
+        onMessagePublished = (req: AtmosphereRequest, res: AtmosphereResponse) => handleMessage(req.responseBody)
+      )
+      Atmosphere.subscribe(atmRequest)
+    } else {
+      isReady = true
+      val atmRequest: AtmosphereRequest = AtmosphereRequest(
```

Comment:
Extract common settings between these two branches and reuse, this is hard to read and even harder to maintain.